### PR TITLE
Enable nightly builds on CircleCI

### PR DESCRIPTION
**Pull request by vincent-psarga**
## Summary

Recently, the build was not working anymore due to a buggy release from one of our dependencies, but that did not get caught by CircleCI as there was no commit done on master.

## Details

CircleCI will build the mono repo once a day, so we should get a notification if that happens again.
I haven't found yet a way to get the parallel workflow (at least without repeating the 127 lines of config for the jobs) but I guess it'd be better in the future to easily pinpoint where the failure occurred.

## Motivation and Context

Better quality and detecting failures with external libraries.

## Types of changes

Not applicable

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -534,6 +534,18 @@ jobs:
 workflows:
   version: 2
 
+  monorepo-nightly:
+    triggers:
+      - schedule:
+          cron: "5 0 * * *"
```
Comment (aslakhellesoy) [inline .circleci/config.yml]: Curious choice of time - five past midnight?

**Comment by aslakhellesoy** [comment]
Please remember to tag issues/PRs

**Comment by charlierudolph** [comment]
Why does a buggy release from a dependency break things? Ideally dependencies are locked down. I know some languages are better at providing tooling for this than others but I would hope all have the ability

**Comment by vincent-psarga** [comment]
On master, the versions of the dependencies are not pinned by `Gemfile.lock`or other equivalent. This allows us to test against the latest releases of the dependencies (as anyway we'll run `make update-dependencies` before the releases).

Also @aslakhellesoy explained me once that on some installs the Gemfile.lock file was not read (for Ruby), maybe when running `gem install` and not using `bundler`

**Comment by luke-hill** [comment]
But the above would only fail due to an external dependency update which would be out of our remit.

The nightly builds would perhaps give us a better timeframe for that. But the reason for the failure would be in the version of something external. Given for an internal dependency change we run CI with over 30 sets of image tests for all the different dependencies (Unless I'm mistaken and we don't run all the permutations?)

**Pull request merged**